sdk/core/Azure.Core/src/Shared/MutableJsonElement.cs

```diff
@@ -48,6 +52,44 @@ internal MutableJsonElement(MutableJsonDocument root, JsonElement element, strin
             }
         }
 
+        internal T ConvertTo<T>()
+        {
+            JsonElement element = GetJsonElement();
+
+            try
+            {
+                // TODO: change this implementation so we don't deserialize known types.
+                // i.e. check typeof(T) and call relevant GetXx() method for supported types.
+                // Can we do better here per perf?
+
+                T? value;
+
+                // Use custom deserialization for an Azure model type.
+                if (typeof(T).GetInterfaces().Contains(typeof(IModelSerializable)) &&
```

Comment:
Can all reflection be in the dynamic layer and the MutableJsonDOcument be all static? Otherwise MJD cannot be used in AOT scenarios.

Reply:
Ah, good call out re: AOT.  I'm not sure offhand why I didn't just use `is` here.  I'll loop back if there was a reason and not just that I was going fast and not thinking.

Reply:
Ah, I see now why I did it this way.  This call needs to call IJsonModelSerializable.Deserialize(), and to do that, it has to have a model instance to populate.  @m-nash, does the serialization infrastructure provide a way to do this without reflection?